docs/source/en/migration.md

```diff
@@ -1 +1,300 @@
+  - The 1st argument is `ctx`, means context, it is the instance of [Context](./basics/extend.md#Context)
+  - The 2nd argument is `next`, use await to execute it for the coming logics.
+- Do not recommend to use `async (ctx, next) => {}`, this is preventing anonymous function in error stack logs 
+- Change `yield next` to `await next()`.
+
+```js
+// 1.x
+module.exports = () => {
+  return function* responseTime(next) {
+    const start = Date.now();
+    yield next;
+    const delta = Math.ceil(Date.now() - start);
+    this.set('X-Response-Time', delta + 'ms');
+  };
+};
+
+// 2.x
+module.exports = () => {
+  return async function responseTime(ctx, next) {
+    const start = Date.now();
+    // Note, differ from the generator function middleware, next is a function, we're executing it here
+    await next();
+    const delta = Math.ceil(Date.now() - start);
+    ctx.set('X-Response-Time', delta + 'ms');
+  };
+};
+```
+
+### yieldable to awaitable
+
+> async was supported in Egg 1.x, thus if the middleware is already async-base, we could skip this section.
+
+[co] supports `yieldable` compatibility types:
+
+- promises
+- array (parallel execution)
+- objects (parallel execution)
+- thunks (functions)
+- generators (delegation)
+- generator functions (delegation)
+
+Despite both `generator` and `async` have the same program models, but we may still need to adjust our codes dependingly after removing `co` because the above special handling from `co`. 
+
+#### promise
+
+We can replace it directly:
+
+```js
+function echo(msg) {
+  return Promise.resolve(msg);
+}
+
+yield echo('hi egg');
+// change to
+await echo('hi egg');
+```
+
+#### array - yield []
+
+`yeild []` is normally used to send concurrent requests, such as:
+
+```js
+const [ news, user ] = yield [
+  ctx.service.news.list(topic),
+  ctx.service.user.get(uid),
+];
+```
+
+In this case, use `Promise.all()` to wrap it:
+
+```js
+const [ news, user ] = await Promise.all([
+  ctx.service.news.list(topic),
+  ctx.service.user.get(uid),
+]);
+```
+
+#### object - yield {}
+
+Sometimes `yield {}` and `yield map` can also be used to send concurrent requests, but it may be a bit complex in this place because `Promise.all` doesn't support Object argument.
+
+```js
+// app/service/biz.js
+class BizService extends Service {
+  * list(topic, uid) {
+    return {
+      news: ctx.service.news.list(topic),
+      user: ctx.service.user.get(uid),
+    };
+  }
+}
+
+// app/controller/home.js
+const { news, user } = yield ctx.service.biz.list(topic, uid);
+```
+
+It's recommended to use `await Promise.all([])`:
+
+```js
+// app/service/biz.js
+class BizService extends Service {
+  list(topic, uid) {
+    return Promise.all([
+      ctx.service.news.list(topic),
+      ctx.service.user.get(uid),
+    ]);
+  }
+}
+
+// app/controller/home.js
+const [ news, user ] = await ctx.service.biz.list(topic, uid);
+```
+
+If the interfaces are unchangeable, we can temporarily do:
```

Comment:
`we can temporarily do` -> `we can do things below as a workaround`

Reply:
Updated